### ISSUE TITLE
Crashed 

### ISSUE THREAD

**Issue by Ravi7489** [googlecolab/colabtools#4842]
**Describe the current behavior**
A clear and concise explanation of what is currently happening.

**Describe the expected behavior**
A clear and concise explanation of what you expected to happen.

**What web browser you are using**
(Chrome, Firefox, Safari, etc.)

**Additional context**
Link to a minimal, public, self-contained notebook that reproduces this issue.
 - Share the file using your GitHub account using  File > Save a copy as a GitHub Gist.
 - or Share Drive notebooks using the Share button then 'Get Shareable Link'.

**Comment by Ravi7489** [comment]
model = keras.Sequential([
    keras.layers.Conv2D(32, (3, 3), activation='relu', input_shape=(28, 28, 1)),
    keras.layers.MaxPooling2D((2, 2)),
    keras.layers.Conv2D(64, (3, 3), activation='relu'),
    keras.layers.MaxPooling2D((2, 2)),
    keras.layers.Conv2D(64, (3, 3), activation='relu'),
    keras.layers.Flatten(),
    keras.layers.Dense(64, activation='relu'),
    keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=5, validation_data=(x_test, y_test))

**Comment by cperry-goog** [comment]
Do you know if this is crashing due to memory limitations? Can you share the error you're seeing?

**Comment by Ravi7489** [comment]
2

I am trying to calibrate my expectations around a single laptop's ability
to train a neural network. I am using tensorflow and keras and after about
say 10 minutes, it crashes. I've seen killsignal 9 exit code 137, and I'm
wondering if this is due to insufficient memory? Other times, when one-hot
encoding using np_utils.to_categorical() I've seen the words memoryerror in
the console, and that's it and my script crashes. This is just trying to
transform the outputs into what a neural net expects before it even runs. I
have 6400 inputs and 1500 outputs and a small hidden layer of 100 nodes.
Batch size 128. That's it. It's not even deep. It crashes whether using an
nvidia gpu or a 4 core cpu. For you pros, is my network too big to train on
my system (i7 4 cores, 16gb ram, nvidia GT 750m , compute capability 3.0).
Is my neural network considered a large one? I have 3 million samples, btw.

1) How do I estimate the amount of memory required for my network? Is it
6400 (# inputs) * 1500 (#outputs) * 4 bytes (per parameter) = 38.4 gb? Can
I see how much memory is being used in real time on a mac somewhere? I've
used activity monitor and the memory pressure gauge is normal. 2) GPUs
typically are maxing out at 8gb-12gb of RAM, whereas CPUs on desktops could
easily have 64 gb. So if the memory requirements of my network exceed 8gb
of RAM, would it be impossible to train on a single GPU? 3) what is the
difference, especially memory wise, between batch_size and batch_training?
Thank you!

**Comment by cperry-goog** [comment]
If you're using a local runtime (you mention your laptop) and if you have questions about how memory works, I'd try stack overflow sorry! If you see a Colab specific bug we can investigate.
